### PR TITLE
Rashuai/ops for rcnn

### PR DESCRIPTION
**Pull request by RandySheriffH**
Implement fixes to convert Mask/Fast RCNN models:

1. Support non-static block_size for Batch2Space and Space2Batch ops;
2. Implement conversion function for CropAndResize;
3. Fix DepthwiseConv with lowered tolerance;
4. Add faster rcnn to pretrained model list for testing.

### CODE REVIEW

File: tf2onnx/onnx_opset/tensor.py [hunk 1]
```diff
@@ -1222,36 +1265,86 @@ def version_1(cls, ctx, node, **kwargs):
...
+            transposed_new_x = ctx.make_node('Transpose', [new_x.output[0]], {'perm': [2, 4, 0, 1, 3, 5]})
+            block_size_prod = ctx.make_node('Mul', [block_size, block_size])
+            const_zero_three = ctx.make_const(utils.make_name(node.name + "_const_zero_three"),
+                                              np.array([0, 3], dtype=np.int64))
+            padded_block_size_prod = ctx.make_node('Pad', [block_size_prod.output[0], const_zero_three.output[0],
+                                                           const_one.output[0]])
+            new_shape_x_v3 = ctx.make_node('Mul', [new_shape_x.output[0], padded_block_size_prod.output[0]])
+            ctx.remove_node(node.name)
+            new_x_v2 = ctx.make_node('Reshape', [transposed_new_x.output[0], new_shape_x_v3.output[0]], name=node.name,
+                                     outputs=node.output)
```
Comment (jignparm) [inline tf2onnx/onnx_opset/tensor.py]: What's the summarized logic here (comments would be helpful)? There are ~20 new nodes -- can they be condensed?

File: tf2onnx/tfonnx.py [hunk 2]
```diff
@@ -90,6 +90,8 @@ def tflist_to_onnx(node_list, shape_override):
                     attr[a] = shape
             elif a == "Tperm":
                 pass
+            elif a == "T_threshold":
```
[the remark this answers is not in the document]
Comment (RandySheriffH) [reply tf2onnx/tfonnx.py]: Removing.

File: tf2onnx/onnx_opset/nn.py [hunk 3]
```diff
@@ -537,6 +537,45 @@ def version_11(cls, ctx, node, **kwargs):
...
+        roi_concat_2 = ctx.make_node("Concat", [const_one_one.output[0], roi_raw_second_half], attr={'axis': 0})
+        final_roi = ctx.make_node("Concat", [roi_concat_1.output[0], roi_concat_2.output[0]], attr={'axis': 0})
+        crop_size_int64 = ctx.make_node("Cast", [crop_size.output[0]], attr={'to': TensorProto.INT64})
+        final_crop_size = ctx.make_node("Concat", [first_half_of_shape, crop_size_int64.output[0]], {'axis': 0})
+        resized_x = ctx.make_node("Resize", [transposed_x.output[0], final_roi.output[0], const_empty_float.output[0],
+                                             final_crop_size.output[0]],
+                                  attr={"mode": mode, "extrapolation_value": extrapolation_value,
+                                        "coordinate_transformation_mode": "tf_crop_and_resize"})
+        ctx.remove_node(node.name)
+        ctx.make_node("Transpose", [resized_x.output[0]], attr={'perm': [0, 2, 3, 1]}, name=node.name,
```
[the remark this answers is not in the document]
Comment (RandySheriffH) [reply tf2onnx/onnx_opset/nn.py]: Done.

File: tf2onnx/onnx_opset/nn.py [hunk 4]
```diff
@@ -537,6 +537,45 @@ def version_11(cls, ctx, node, **kwargs):
...
+    @classmethod
+    def version_11(cls, ctx, node, **kwargs):
+        # crop by roialign then resize
+        mode = "nearest" if node.attr["method"] is not None and node.attr["method"].s == b"nearest" else "linear"
+        extrapolation_value = float(node.attr["extrapolation_value"].f) if node.attr[
+                                                                               "extrapolation_value"] is not None else 0.0
+        input_x = node.inputs[0]
+        boxes = node.inputs[1]
+        crop_size = node.inputs[3]
+        transposed_x = ctx.make_node("Transpose", [input_x.output[0]], attr={'perm': [0, 3, 1, 2]})
```
[the remark this answers is not in the document]
Comment (RandySheriffH) [reply tf2onnx/onnx_opset/nn.py]: Done.

File: tf2onnx/onnx_opset/nn.py [hunk 5]
```diff
@@ -537,6 +537,45 @@ def version_11(cls, ctx, node, **kwargs):
         cls.version_1(ctx, node, **kwargs)
 
 
+@tf_op(["CropAndResize"])
+class CropAndResize:
+    @classmethod
+    def version_11(cls, ctx, node, **kwargs):
+        # crop by roialign then resize
+        mode = "nearest" if node.attr["method"] is not None and node.attr["method"].s == b"nearest" else "linear"
+        extrapolation_value = float(node.attr["extrapolation_value"].f) if node.attr[
```
[the remark this answers is not in the document]
Comment (RandySheriffH) [reply tf2onnx/onnx_opset/nn.py]: Done.

File: tf2onnx/onnx_opset/nn.py [hunk 6]
```diff
@@ -537,6 +537,45 @@ def version_11(cls, ctx, node, **kwargs):
         cls.version_1(ctx, node, **kwargs)
 
 
+@tf_op(["CropAndResize"])
+class CropAndResize:
+    @classmethod
+    def version_11(cls, ctx, node, **kwargs):
+        # crop by roialign then resize
+        mode = "nearest" if node.attr["method"] is not None and node.attr["method"].s == b"nearest" else "linear"
```
[the remark this answers is not in the document]
Comment (RandySheriffH) [reply tf2onnx/onnx_opset/nn.py]: Done.

File: tf2onnx/onnx_opset/tensor.py (hunk 1 above)
Comment (RandySheriffH) [reply tf2onnx/onnx_opset/tensor.py]: Commented.

File: tf2onnx/onnx_opset/nn.py [hunk 7]
```diff
@@ -537,6 +537,71 @@ def version_11(cls, ctx, node, **kwargs):
...
+    def version_11(cls, ctx, node, **kwargs):
+        # create loop of resize to cater to tensorflow CropAndResize, one box one iteration
+        mode = "nearest" if node.get_attr("method") is not None and node.get_attr(
+            "method").s == b"nearest" else "linear"
+        extrapolation_value = float(node.get_attr("extrapolation_value", "0").f)
+        input_x = node.inputs[0]
+        boxes = node.inputs[1]
+        box_ind = node.inputs[2]
+        crop_size = node.inputs[3]
+        trip_name = utils.make_name("i")
```
[the remark this answers is not in the document]
Comment (RandySheriffH) [reply tf2onnx/onnx_opset/nn.py]: Done.

**Pull request closed without merging**